src/compiler/Restler.Compiler/Dependencies.fs

```diff
@@ -1247,10 +1265,7 @@ let extractDependencies (requestData:(RequestId*RequestData)[])
     logTimingInfo "Getting producers..."
 
     requestData
-    // Only include POST, PUT, PATCH, and GET requests.  Others are never producers.
-    |> Array.filter (fun (r, _) -> [ OperationMethod.Post ; OperationMethod.Put; OperationMethod.Patch ;
-                                     OperationMethod.Get ]
-                                    |> List.contains r.method)
```

Comment:
Could we please merge this and the corresponding tests separately, since this is a separate bug fix that could have wide implications on dependencies?  I'm curious why this change is needed, since the soft delete scenario has been working for me with global annotations without this change (using the annotation below).
```
    {
      "producer_endpoint": "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/vaults/{vaultName}",
      "producer_method": "DELETE",
      "producer_resource_name": "vaultName",

      "consumer_endpoint": "/subscriptions/{subscriptionId}/providers/Microsoft.KeyVault/locations/{location}/deletedVaults/{vaultName}",
      "consumer_method": "GET",
      "consumer_param": "vaultName"
    
```

Reply:
Update: I haven't tested yet but I believe if you add a PUT of the config store into the soft_delete.json specs, this change should not be needed.

Reply:
I tried reverting this change and it caused my test to fail. I think this is because links are specified in the _producer_ operation rather than in the consumer, as they are for RESTler annotations. I think I have a good fix -- I just broke out the handling of links into a separate code block.

Reply:
I am going to debug this a bit more to understand the root cause of the failure with the added PUT (currently not getting the expected behavior as well).  It would be good to avoid grammars that have a DELETE as a producer (without a corresponding PUT or POST first) for the time being, since the engine makes some assumptions on the producer methods as well.

Reply:
I should clarify -- I did not try adding a PUT to my test. I only verified that the test as it exists (without a PUT) failed if DELETE was excluded from the potential producers.

And I don't understand the rationale for excluding DELETE from being a producer, and I don't recall seeing any documentation to this effect. Can you say more about what assumptions the engine makes about this?